GVFS/GVFS.Common/ProductUpgrader.cs

```diff
@@ -84,10 +84,20 @@ public bool ReleaseUpgradeLock()
         {
             List<Release> releases;
 
-            if (this.TryLoadRingConfig(out errorMessage) && 
-                this.TryFetchReleases(out releases, out errorMessage))
+            newVersion = null;
+            if (!this.TryLoadRingConfig(out errorMessage))
+            {
+                return false;
+            }
+
+            if (this.Ring == RingType.None)
+            {
+                errorMessage = "Upgrade ring set to None. No upgrade check was performed.";

```

Comment:
Why is this treated as an error? What does the user see as a result of this?

Reply:
This is what the user sees -

PS M:\Work\GVFS\VFSForGit> gvfs upgrade
Checking for GVFS upgrades...Failed
Upgrade ring set to None. No upgrade check was performed.
PS M:\Work\GVFS\VFSForGit>

Reply:
I think what I would expect is either no spinner at all and a success message about None or just being up to date; or if you want to show the spinner, then it should end in Succeeded and a similar message.

There's really no need for a spinner at all if we already know we're going to noop, so I would personally favor the first approach. This could be rolled into the prereq checks, right?

Reply:
- Spinner removed.
- This is not treated as error anymore.
- New messaging (the message text can be reviewed and updated in a future PR)

PS M:\Work\GVFS\VFSForGit> gvfs upgrade
Upgrade ring set to None. No upgrade check was performed.
PS M:\Work\GVFS\VFSForGit>